### PR TITLE
Fix ModuleUtils::renameFunctions on RefFunc

### PR DESCRIPTION
**Pull request by kripken**
It had hardcoded handling of the table, but was missing RefFunc (and maybe more?)

Also rewrite it to be clearer (avoid special-casing the table) and faster (parallel).

### CODE REVIEW

**Comment by kripken** [comment]
Not really possible to test this atm as it's only used in Asyncify, and on functions that can't appear in the table (the special "runtime" functions in the Asyncify API). But if we use this in the future, we might hit the bug.

File: src/ir/module-utils.h [hunk 1]
```diff
@@ -186,30 +186,34 @@ template<typename T> inline void renameFunctions(Module& wasm, T& map) {
...
+
+  // Update all references to it.
+  struct Updater : public WalkerPass<PostWalker<Updater>> {
+    bool isFunctionParallel() override { return true; }
+
+    T& map;
+
+    void maybeUpdate(Name& name) {
+      auto iter = map.find(name);
+      if (iter != map.end()) {
```
Comment (tlively) [inline src/ir/module-utils.h:198]: We can use C++17 fanciness to make this more pithy:
```suggestion
      if (auto iter = map.find(name); iter != map.end()) {
```
(Also we typically use `it` for iterators like this)

**Review by tlively — APPROVED** [review]
Can't comment inline, but there is a "to" missing in the last line of the comment above `renameFunctions`. Also the variable `F` should be renamed to `f`.

If we really want to test this, we could in principle write a gtest for it. But I'd be ok landing this without a test.

File: src/ir/module-utils.h (hunk 1 above)
Comment (kripken) [reply src/ir/module-utils.h:198]: About `it` vs `iter`, either sgtm, but seems like `iter` is the current majority:
```
$ grep "auto iter =" `find -name "*.cpp"` | wc -l
94
$ grep "auto it =" `find -name "*.cpp"` | wc -l
55
```

Comment (tlively) [reply src/ir/module-utils.h]: oh interesting. I guess it's only the standard in code I frequently look at 😬

**Pull request merged**